medusa/config.py

```diff
@@ -652,6 +676,46 @@ def check_setting_str(config, cfg_name, item_name, def_val, silent=True, censor_
 
 
 ################################################################################
+# Check_setting_list                                                           #
+################################################################################
+def check_setting_list(config, cfg_name, item_name, default=None, censor_log=False, transform=None, transform_default=0):
+    """Check a setting, using the settings section and item name. Expect to return a list."""
+    default = default or []
+    censor_log = False
+
+    if not censor_log:
+        censor_level = common.privacy_levels['stupid']
+    else:
+        censor_level = common.privacy_levels[censor_log]
+    privacy_level = common.privacy_levels[app.PRIVACY_LEVEL]
+
+    try:
+        my_val = config[cfg_name][item_name]
+    except Exception:
+        my_val = default
+        try:
+            config[cfg_name][item_name] = my_val
+        except Exception:
+            config[cfg_name] = {}
+            config[cfg_name][item_name] = my_val
+
+    if privacy_level >= censor_level or (cfg_name, item_name) in iteritems(logger.censored_items):
+        if not item_name.endswith('custom_url'):
+            logger.censored_items[cfg_name, item_name] = my_val
```

Comment:
For now it's only used to sensor some tokens. Also your example won't work. As it will overwrite the value for n items in the list.